lib/ansible/module_utils/kubevirt.py

```diff
@@ -24,20 +24,16 @@
 
 
 VM_COMMON_ARG_SPEC = {
-    'name': {},
+    'name': {'required': True},
+    'namespace': {'required': True},
     'force': {
         'type': 'bool',
         'default': False,
     },
     'resource_definition': {
-        'type': list_dict_str,
+        'type': dict,
         'aliases': ['definition', 'inline']
     },
-    'src': {
-        'type': 'path',
-    },
-    'namespace': {},
-    'api_version': {'type': 'str', 'default': '%s/%s' % (API_GROUP, MAX_SUPPORTED_API_VERSION), 'aliases': ['api', 'version']},
```

Comment:
I think we should keep this for user, mainly in this early stage.

Reply:
I thought about it, but wasn't able to come up with a valid use case for this, as:

1. You can't force using something newer than `MAX_SUPPORTED_API_VERSION`, as current code will reject that.
2. If your cluster has api_ver lower than max_supported, then it'll get used automatically anyway.
3. Afaik there are no scenarios possible where two separate deployments (with different versions) 
of kubevirt are present on a cluster and you pick the one you want with this parameter.

And those are all the scenarios I could come up with. So as far as I can tell there is really no way to actually use `api_version` for anything.

Reply:
Well, I think MAX_SUPPORTED_API_VERSION should be overriden by this parameter

Reply:
Ok, but that's a separate patch and a separate discussion. For now `api_version` does nothing, which is why I've removed it.

Reply:
So we will remove it, and add it back? Okay :)

Reply:
It makes sense from a release process perspective. It'd be nice if our modules looked polished (and between the two of us, I'm more of an expert in polish(ing) things ;) in 2.8, which is just around the corner, and this patch does that. This way we're guaranteed to not release the modules with a parameter that does nothing.

Should we have a discussion about adding `api_version` that actually overrides MAX_VERSION? Probably. However:
1. I'm not sure we'll do that before 2.8 as we do have other priorities.
2. I'm not sure how I feel about it. (And you don't know how Piotr feels about it.)
3. Thinking about it now, if it does get added, I'm not sure it should be called `api_version`, but something else instead that makes it explicit that using that parameter is risky.

Etc., etc. Bottom line – cleanup first to make sure we look good in 2.8, then worry about *proper* ways of adding stuff.

Reply:
My point is, that I don't want to update the modules every time new API version is released, if it don't break *our* backward compatibility. So it would be really nice to have the possibility to work with modules even with `v1beta3` again if it don't break *our* compatibility. So either we should have this parameter here, as we can't add it after 2.8 is released or we should change the login in `find_supported_resource` method.